docs/cypress/e2e/links.cy.ts

```diff
@@ -0,0 +1,136 @@
+import { VALIDATED_LINKS } from '../data/validatedLinks';
+import { REQUEST_GET_LINKS } from '../data/requestGetLinks';
+import { BASE_URL } from '../data/constants';
+
+let allLinks: string[] = [];
+const numberOfLinks = 119;
+const requestedLinks: Set<string> = new Set();
+
+before(() => {
+  cy.task('readSitemapLinks').then((links: string[]) => {
+    allLinks = allLinks.concat(links);
+  });
+});
+
+describe('Local Sitemap', () => {
+  it('should have 119 links', () => {
+    expect(allLinks.length).to.eq(numberOfLinks);
+  });
+});
+
+describe(`All pages on Sitemap`, () => {
+  for (let i = 0; i < numberOfLinks; i++) {
+    it(`all links on page ${i} should work`, () => {
+      const link = allLinks[i];
+      cy.task('log', `🧪[TESTING...] page ${BASE_URL}/${link}`);
+      cy.visit(link || '/');
+      cy.get('a').each(hrefWorks);
+
+      function hrefWorks(htmlTag: JQuery<HTMLElement>): void {
+        const tagHref: string = htmlTag.prop('href');
+        const tagHash: string = htmlTag.prop('hash');
+        const tagText: string = htmlTag.prop('text');
+        const tagName: string = htmlTag.prop('tagName');
+        let pureHref: string;
+        if (tagHref) {
+          pureHref = tagHref.replace(tagHash, ''); // TODO: add test to validate links with a hash tag.
+          logMessage('CHECKING');
+
+          if (allLinks.includes(`${pureHref.replace(`${BASE_URL}/`, '')}`)) {
+            expect(`${pureHref.replace(`${BASE_URL}/`, '')}`).to.oneOf(
+              allLinks
+            );
+            logMessage('SKIPPING_SITEMAP');
+          } else if (
+            allLinks.includes(`${pureHref.replace(`${BASE_URL}/`, 'react/')}`)
+          ) {
+            expect(`${pureHref.replace(`${BASE_URL}/`, 'react/')}`).to.oneOf(
+              allLinks
+            );
+            logMessage('SKIPPING_SITEMAP');
+          } else if (
+            VALIDATED_LINKS.includes(pureHref) ||
+            VALIDATED_LINKS.includes(`${pureHref.replace(BASE_URL, '')}`) ||
+            requestedLinks.has(pureHref)
+          ) {
+            logMessage('SKIPPING_VALIDATED');
+          } else {
+            const requestMethod = REQUEST_GET_LINKS.includes(pureHref)
+              ? 'GET'
+              : 'HEAD';
+            logMessage('REQUESTING');
+            requestedLinks.add(pureHref);
+            cy.request({
+              url: pureHref,
+              followRedirect: false,
+              method: requestMethod,
+            }).then(({ status }) => {
+              logMessage('RETURNING', status);
+              expect(status).to.oneOf([200, 301, 303]);
+              cy.clearLocalStorage();
+            });
+          }
+        } else if (tagName === 'A') {
+          logMessage('NO_HREF');
+        }
+
+        type EvtName =
+          | 'CHECKING'
+          | 'SKIPPING_SITEMAP'
+          | 'SKIPPING_VALIDATED'
+          | 'REQUESTING'
+          | 'RETURNING'
+          | 'NO_HREF';
```

Comment:
Why are we declaring types within the body of a function?

Reply:
Because we don't reuse the type, defining it close to where it's used might be the cleanest way.

Reply:
Types should be declared in a _types.ts_ files or at the top of a file

Reply:
Done. Moved to top.